app/lib/actions/pulp/repository/remove_units.rb

```diff
@@ -0,0 +1,51 @@
+module Actions
+  module Pulp
+    module Repository
+      class RemoveUnits < Pulp::AbstractAsyncTask
+        input_format do
+          param :repo_id
+          param :contents
+          param :content_unit_type
+        end
+
+        def invoke_external_task
+          repo = ::Katello::Repository.find_by(:id => input[:repo_id])
+          if repo.nil?
+            repo = ::Katello::ContentViewPuppetEnvironment.find_by(:id => input[:repo_id])
+            repo = repo.nonpersisted_repository
+          end
+          repo_content_types = ::Katello::RepositoryTypeManager.find(repo.content_type).content_types
+          tasks = []
+          if input[:contents]
+            if input[:content_unit_type]
+              content_type = ::Katello::RepositoryTypeManager.find_content_type(input[:content_unit_type].downcase)
+              units = content_type.model_class.where(:id => input[:contents])
+              unit_pulp_ids = units.map(&:pulp_id)
+              tasks << ::SmartProxy.pulp_master.content_service(content_type).remove(repo, unit_pulp_ids)
+            else
+              user_removable_content_types = ::Katello::RepositoryTypeManager.find(repo.content_type).user_removable_content_types
+              user_removable_content_types.each do |user_removable_content_type|
+                units = user_removable_content_type.model_class.where(:id => input[:contents])
+                unit_pulp_ids = units.map(&:pulp_id)
+                tasks << ::SmartProxy.pulp_master.content_service(user_removable_content_type).remove(repo, unit_pulp_ids) unless unit_pulp_ids.blank?
+              end
```

Comment:
Is this condition actually valid?  Could you pass a bunch of ids but no type?   I'm not sure it is (in which case we should probably just throw an error, rather than trying to remove all unit types).

Reply:
Also, would it be possible to break this method up a bit or simplify it?